service/controller/resource/templateunits/create.go

```diff
@@ -4,17 +4,21 @@ import (
 	"context"
 
 	"github.com/giantswarm/microerror"
-	"github.com/shurcooL/httpfs/vfsutil"
 
-	"github.com/giantswarm/ignition-operator/data"
+	"github.com/giantswarm/ignition-operator/service/controller/controllercontext"
+	"github.com/giantswarm/ignition-operator/service/controller/key"
 )
 
 func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
-	b, err := vfsutil.ReadFile(data.Assets, "worker_template.yaml")
+	cc, err := controllercontext.FromContext(ctx)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	cc.Status.Units, err = key.Render(cc.Spec, key.UnitPath)
```

Comment:
Why templating units and files is split at all?

Reply:
I wanted to have this split because it makes the rendering process more understandable.
Files and units are two sections in the final ignition. Splitting the rendering for these makes it easier to template the final ignition as there are two maps.
Additionally my idea was that this makes it easy to extend: There can be just a unit or just a file added from the outside later on.